regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/helpers/BranchTrackingVisitor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SonarSource Analyzers Regex Parsing Commons
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons.regex.helpers;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import org.sonarsource.analyzer.commons.regex.ast.AbstractRegexSyntaxElement;
+import org.sonarsource.analyzer.commons.regex.ast.DisjunctionTree;
+import org.sonarsource.analyzer.commons.regex.ast.IndexRange;
+import org.sonarsource.analyzer.commons.regex.ast.RegexBaseVisitor;
+import org.sonarsource.analyzer.commons.regex.ast.RegexTree;
+import org.sonarsource.analyzer.commons.regex.ast.RepetitionTree;
+
+/**
+ * The BranchTrackingVisitor saves the latest branching nodes as it traverses the tree. This is useful to avoid
```

Comment:
I think this description could be more clear. Until your fix it wasn't clear to me that the old behavior wasn't what you had in mind when you said "latest branching node". I think something like "nearest *enclosing* branching construct" would capture what you mean more closely.

Reply:
Agreed, I changed it.